File: nakano-death/convert.js
const fs = require('fs');

const geojson = JSON.parse(fs.readFileSync('nakano.json', 'utf8'));
const features = geojson.features;

// ID Mapping
const idMap = {
    "Kamitakada": "kamitakada",
    "Higashinakano": "higashinakano",
    "Chuo": "chuo",
    "Honcho": "honcho",
    "Yayoicho": "yayoimachi", // Remap to match config.js
    "Minamidai": "minamidai",
    "Nakano": "nakano",
    "Kamisaginomiya": "kamisaginomiya",
    "Saginomiya": "saginomiya",
    "Eharacho": "eharacho",
    "Egoda": "egota", // Remap to match config.js
    "Matsugaoka": "matsugaoka",
    "Maruyama": "maruyama",
    "Shirasagi": "shirasagi",
    "Wakamiya": "wakamiya",
    "Yamatocho": "yamatocho", // Note: Yamatocho vs Yamato-cho
    "numabukuro": "numabukuro", // Lowercase in JSON?
    "Numabukuro": "numabukuro", // Just in case
    "Arai": "arai",
    "Nogata": "nogata"
};

// Calculate Bounding Box
let minLon = Infinity, maxLon = -Infinity;
let minLat = Infinity, maxLat = -Infinity;

features.forEach(f => {
    // Handle Polygon and MultiPolygon
    const coords = f.geometry.type === 'MultiPolygon' ? f.geometry.coordinates.flat() : f.geometry.coordinates;
    coords.forEach(ring => {
        ring.forEach(pt => {
            const [lon, lat] = pt;
            if (lon < minLon) minLon = lon;
            if (lon > maxLon) maxLon = lon;
            if (lat < minLat) minLat = lat;
            if (lat > maxLat) maxLat = lat;
        });
    });
});

console.log(`Bounds: Lon[${minLon}, ${maxLon}], Lat[${minLat}, ${maxLat}]`);

// Setup SVG ViewBox
const width = 600;
const height = 800; // Target size

const lonSpan = maxLon - minLon;
const latSpan = maxLat - minLat;

// Determine scale to fit
const scaleX = width / lonSpan;
const scaleY = height / latSpan;
const scale = Math.min(scaleX, scaleY) * 0.95; // 95% fit to leave some margin

console.log(`Scale: ${scale}`);

// Calculate logic offsets to center it
const mkWidth = lonSpan * scale;
const mkHeight = latSpan * scale;
const offsetX = (width - mkWidth) / 2;
const offsetY = (height - mkHeight) / 2;

const mapPaths = {};

features.forEach(f => {
    let rawName = f.properties.CITY_NAME;
    let id = idMap[rawName] || rawName.toLowerCase();

    // Check if Numabukuro is lowercase in JSON source (it appeared as "numabukuro" in snippets)

    let pathStr = "";

    const polygons = f.geometry.type === 'MultiPolygon' ? f.geometry.coordinates : [f.geometry.coordinates];

    polygons.forEach(poly => {
        poly.forEach((ring, index) => {
            // Move to first point
            const pts = ring.map(pt => {
                const [lon, lat] = pt;
                const x = (lon - minLon) * scale + offsetX;
                // SVG Y is Top-Down, Lat is Bottom-Up -> Invert Lat
                const y = height - ((lat - minLat) * scale + offsetY);
                return `${x.toFixed(1)},${y.toFixed(1)}`;
            });

            pathStr += "M " + pts.join(" L ") + " Z ";
        });
    });

    mapPaths[id] = pathStr.trim();
});

console.log("const mapPaths = " + JSON.stringify(mapPaths, null, 4) + ";");

File: nakano-death/convert.py
import json
import math

with open('nakano.json', 'r') as f:
    geojson = json.load(f)

features = geojson['features']

# ID Mapping
id_map = {
    "Kamitakada": "kamitakada",
    "Higashinakano": "higashinakano",
    "Chuo": "chuo",
    "Honcho": "honcho",
    "Yayoicho": "yayoimachi", 
    "Minamidai": "minamidai",
    "Nakano": "nakano",
    "Kamisaginomiya": "kamisaginomiya",
    "Saginomiya": "saginomiya",
    "Eharacho": "eharacho",
    "Egoda": "egota", 
    "Matsugaoka": "matsugaoka",
    "Maruyama": "maruyama",
    "Shirasagi": "shirasagi",
    "Wakamiya": "wakamiya",
    "Yamatocho": "yamatocho",
    "numabukuro": "numabukuro",
    "Numabukuro": "numabukuro",
    "Arai": "arai",
    "Nogata": "nogata"
}

# Calculate Bounding Box
min_lon = float('inf')
max_lon = float('-inf')
min_lat = float('inf')
max_lat = float('-inf')

for f in features:
    geom = f['geometry']
    coords = geom['coordinates']
    if geom['type'] == 'MultiPolygon':
        # List of Polygons -> List of Rings -> List of Points
        for poly in coords:
            for ring in poly:
                for pt in ring:
                    lon, lat = pt
                    min_lon = min(min_lon, lon)
                    max_lon = max(max_lon, lon)
                    min_lat = min(min_lat, lat)
                    max_lat = max(max_lat, lat)
    else:
         for ring in coords:
                for pt in ring:
                    lon, lat = pt
                    min_lon = min(min_lon, lon)
                    max_lon = max(max_lon, lon)
                    min_lat = min(min_lat, lat)
                    max_lat = max(max_lat, lat)

print(f"Bounds: Lon[{min_lon}, {max_lon}], Lat[{min_lat}, {max_lat}]")

# Setup SVG ViewBox
width = 600
height = 800

lon_span = max_lon - min_lon
lat_span = max_lat - min_lat

# Determine scale to fit
scale_x = width / lon_span
scale_y = height / lat_span
scale = min(scale_x, scale_y) * 0.95 

# Calculate offsets to center
mk_width = lon_span * scale
mk_height = lat_span * scale
offset_x = (width - mk_width) / 2
offset_y = (height - mk_height) / 2

map_paths = {}

for f in features:
    raw_name = f['properties'].get('CITY_NAME')
    if not raw_name: # Handle lowercase keys if present
         raw_name = f['properties'].get('city_name')
         
    id_key = id_map.get(raw_name, raw_name.lower())
    
    path_str = ""
    geom = f['geometry']
    coords = geom['coordinates']
    polygons = coords if geom['type'] == 'MultiPolygon' else [coords]
    
    for poly in polygons:
        for ring in poly:
            pts_str = []
            for pt in ring:
                lon, lat = pt
                x = (lon - min_lon) * scale + offset_x
                # Invert Y
                y = height - ((lat - min_lat) * scale + offset_y)
                pts_str.append(f"{x:.1f},{y:.1f}")
            
            path_str += "M " + " L ".join(pts_str) + " Z "

    map_paths[id_key] = path_str.strip()

print("const mapPaths = " + json.dumps(map_paths, indent=4, ensure_ascii=False) + ";")
